bowconvert.go

```diff
@@ -130,6 +138,73 @@ func ToString(input interface{}) (output string, ok bool) {
 		return fmt.Sprintf("%f", input), true
 	case float64:
 		return fmt.Sprintf("%f", input), true
+	case arrow.Timestamp:
+		return strconv.Itoa(int(input)), true
+	}
+	return
+}
+
+// ToTimestampSec returns an arrow.Timestamp value and a bool whether the conversion was successful or not.
+// String values are first interpreted with strconv.ParseInt.
+// If it fails, the values are parsed with arrow.TimestampFromString with the arrow.Second time unit.
+func ToTimestampSec(input interface{}) (output arrow.Timestamp, ok bool) {
+	return toTimestamp(input, arrow.Second)
+}
+
+// ToTimestampMilli returns an arrow.Timestamp value and a bool whether the conversion was successful or not.
+// String values are first interpreted with strconv.ParseInt.
+// If it fails, the values are parsed with arrow.TimestampFromString with the arrow.Millisecond time unit.
+func ToTimestampMilli(input interface{}) (output arrow.Timestamp, ok bool) {
+	return toTimestamp(input, arrow.Millisecond)
+}
+
+// ToTimestampMicro returns an arrow.Timestamp value and a bool whether the conversion was successful or not.
+// String values are first interpreted with strconv.ParseInt.
+// If it fails, the values are parsed with arrow.TimestampFromString with the arrow.Microsecond time unit.
+func ToTimestampMicro(input interface{}) (output arrow.Timestamp, ok bool) {
+	return toTimestamp(input, arrow.Microsecond)
+}
+
+// ToTimestampNano returns an arrow.Timestamp value and a bool whether the conversion was successful or not.
+// String values are first interpreted with strconv.ParseInt.
+// If it fails, the values are parsed with arrow.TimestampFromString with the arrow.Nanosecond time unit.
+func ToTimestampNano(input interface{}) (output arrow.Timestamp, ok bool) {
+	return toTimestamp(input, arrow.Nanosecond)
+}
+
+func toTimestamp(input interface{}, timeUnit arrow.TimeUnit) (output arrow.Timestamp, ok bool) {
+	switch input := input.(type) {
+	case json.Number:
+		output, err := input.Int64()
+		return arrow.Timestamp(output), err == nil
+	case int:
+		return arrow.Timestamp(input), true
```

Comment:
We are doing exactly the same thing in many cases, could we factor them ?

Reply:
The line `return arrow.Timestamp(input), true` is repeated 7 times. I think we can factorize them in order to avoid future errors when something will change here and we will have forgotten to update one or two lines

Reply:
I'm open to suggestions, but I don't see any way to factorize them...

Reply:
Also, this is out of the scope of this PR, which is only adding support for new types.

Reply:
We can do:

`
case int, int8, int16, int32, int64, float32, float64:
    return arrow.Timestamp(input), true
`

Reply:
This is a type switch that permits several type assertions in series.
This is like a regular switch statement, but the cases specify types (not values), and those values are compared against the type of the value held by the given interface value.
This switch statement tests whether the interface value `input` holds a value of type `int`, `int8`, etc.
In each cases, the asserted variable `input` will be of type `int`, `int8`, etc. respectively and hold the value held by `input`. In the default case (where there is no match), or with multiple cases (your suggestion here), the asserted variable `input` is of the same interface type and value as `input`.
As we cannot convert an interface value to arrow.Timestamp, this doesn't work :'(

Reply:
strange, my guess was as good as @yoyocm!
If you tried let's stay this way, there are several lazy type conversion in go out in the wild and they generally look like this.

Reply:
Sorry, I don't get it, what do you suggest exactly?